From: Ben Hutchings <ben@decadent.org.uk>
Date: Tue, 24 Feb 2015 09:53:48 +0000
Subject: perf: Fix ABI change in 3.16.7-ckt6
Forwarded: not-needed

Commit c3c87e770458 ('perf: Tighten (and fix) the grouping condition')
removed perf_event_context::type and its type definition.  Add them
back.

--- a/include/linux/perf_event.h
+++ b/include/linux/perf_event.h
@@ -453,6 +453,12 @@ struct perf_event {
 #endif /* CONFIG_PERF_EVENTS */
 };
 
+/* now unused */
+enum perf_event_context_type {
+	task_context,
+	cpu_context,
+};
+
 /**
  * struct perf_event_context - event context structure
  *
@@ -460,6 +466,7 @@ struct perf_event {
  */
 struct perf_event_context {
 	struct pmu			*pmu;
+	enum perf_event_context_type	type; /* now unused */
 	/*
 	 * Protect the states of the events in the list,
 	 * nr_active, and the list:
